WordPress/gutenberg
# Design Tools: Add a Position block support (including sticky), decoupled from layout

## Related

* https://github.com/WordPress/gutenberg/issues/30121
* https://github.com/WordPress/gutenberg/pull/44723
* https://github.com/WordPress/gutenberg/discussions/46032
* Will require the following to fix placement of toolbar controls while a block is selected: https://github.com/WordPress/gutenberg/pull/46565

## What?


Fixes: #30121 (note this PR is still a WIP, so the "fixes" label here is simply to associate this PR with the open issue)

This PR is an alternative to https://github.com/WordPress/gutenberg/pull/44723, and adds support for sticky position, decoupled from the Layout panel and whether or not a block has layout support. It explores ideas from the discussion in https://github.com/WordPress/gutenberg/discussions/46032

## Why?


Sticky / fixed position will be valuable for sticky headers and footers, and creative pattern applications. However, it is likely to be used infrequently, and needs to be de-emphasised in the UI. This PR is an exploration of the ideas in https://github.com/WordPress/gutenberg/discussions/46032, to see what the most minimal version of the feature might look like.

## How?


* Take the ideas from #44723 and implement as a separate position block support
* Introduce the Position controls (design still to be finalised) under a separate Position panel in the inspector controls
* Opt the Group block in to using sticky position

## Testing Instructions





Note: sticky position will stick to the top of the direct parent block. If at first it appears not to be working, ensure that the parent block is tall enough / has enough content to be scrolled.

1. Open up the block editor, add a group block and put a paragraph inside it, and give the group block a background color
2. In the inspector controls, under Position, set the position to Sticky
3. Scroll down, and the block should stick to the top of the screen
4. Publish the post or page, and on the site frontend, the block should stick to the top as you scroll, and factor in the height of the logged-in admin bar when it is visible

## Screenshots or screencast 

The position control situated under Position:

<img width="1115" alt="image" src="https://user-images.githubusercontent.com/14988353/205223008-57acb82c-3788-400c-b505-0af13cdb9883.png">

A screengrab of selecting the control from the Position panel:

![2022-12-02 16 34 11](https://user-images.githubusercontent.com/14988353/205222613-e422c3c1-7825-4eee-9d20-ece5217ca02a.gif)
